### DISCUSSION TITLE
How is the swr cache implemented and what technology is used?

### DISCUSSION THREAD

**Discussion by qiaogaolong** [vercel/swr#2233] [Q&A]
How is the swr cache implemented and what technology is used?

**Comment by dev-afzalansari** [comment]
@qiaogaolong,

**SWR** uses a **Cache Provider** which should must have javascript's [Map Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) like **apis** to get, set and delete the cache and it uses **Map Object** as it's default **Cache Provider** until you override it by providing your own **Cache Provider**.

Learn more about [Caching in swr](https://swr.vercel.app/docs/advanced/cache)

**Reply by qiaogaolong** [reply]
Hello, I just don't know where the data after the interface request is cached

**Reply by qiaogaolong** [reply]
At present, after the interface request, the content display in the cache will be obtained first, but it is not clear where the data is cached.

**Reply by qiaogaolong** [reply]
![image](https://user-images.githubusercontent.com/35677796/200264641-a799b246-61f4-4d21-964d-63cf8a19b5d1.png)

How to achieve the above documentation description? Where exactly is it stored? I can not find

**Reply by dev-afzalansari** [reply]
> Where exactly is it stored?

It's stored in the Map Object.

Please make yourself familiar with [Map Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

**Reply by qiaogaolong** [reply]
ok, i'll find out, thanks

**Reply by qiaogaolong** [reply]
Thank you, got it, I thought it was stored in the cache哈哈😂
